runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsPlugin.java

```diff
@@ -220,11 +235,35 @@ public void onGameStateChanged(GameStateChanged event)
 	{
 		if (event.getGameState() == GameState.LOADING)
 		{
+			wasInCrypt = isInCrypt();
 			// on region changes the tiles get set to null
 			walls.clear();
 			ladders.clear();
 			puzzleAnswer = null;
 		}
+		else if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			if (wasInCrypt && !isInCrypt())
+			{
+				prayerDrainTimer.stop();
+			}
+			else if (!wasInCrypt && isInCrypt())
+			{
+				if (prayerDrainTimer == null)
+				{
+					prayerDrainTimer = new BarrowsPrayerDrainTimer(this, spriteManager);
```

Comment:
Is there another place I can initialize this so I don't have to mark it as Nullable and lazily initialize it? I tried to do it in startUp, but that resulted in the spriteManager being unable to get the image.